internal/policy/policy_test.go

```diff
@@ -337,3 +336,95 @@ func TestGetStateForCommit(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, firstState, state)
 }
+
+func TestFindDelegationEntry(t *testing.T) {
+	t.Run("no delegation", func(t *testing.T) {
+		state := createTestStateWithPolicy(t)
+
+		entry, err := state.findDelegationEntry("random")
+		assert.ErrorIs(t, err, ErrDelegationNotFound)
+		assert.Equal(t, entry, tuf.Delegation{})
+	})
+	t.Run("simple delegation", func(t *testing.T) {
+		state := createTestStateWithPolicy(t)
```

Comment:
I wonder if we can have a helper like this one `createTestStateWithDelegatedPolicies`.

Reply:
That would do away with all the temporary policies that manually modify `State`. Also, you could have actual policies there that we could use in other tests, say to verify signatures using delegated policies.

Reply:
Fixed